sunpy/physics/differential_rotation.py

```diff
@@ -87,317 +90,94 @@ def diff_rot(duration, latitude, rot_type='howard', frame_time='sidereal'):
-        - 0.000140 * np.cos(np.deg2rad(2.0 * me)) \
-        + 0.0000160 * np.cos(np.deg2rad(58.30 + 2.0 * mv - 2.0 * me)) \
-        + 0.0000050 * np.cos(np.deg2rad(209.10 + mv - me)) \
-        + 0.0000050 * np.cos(np.deg2rad(253.80 - 2.0 * mm + 2.0 * me)) \
-        + 0.0000160 * np.cos(np.deg2rad(89.50 - mj + me)) \
-        + 0.0000090 * np.cos(np.deg2rad(357.10 - 2.0 * mj + 2.0 * me)) \
-        + 0.0000310 * np.cos(np.deg2rad(d))
-
-    sd_const = constants.radius / constants.au
-    sd = np.arcsin(sd_const / r) * 10800.0 / np.pi
-
-    return {"p": Angle(p, u.deg),
-            "b0": Angle(b, u.deg),
-            "sd": Angle(sd.value, u.arcmin),
-            "l0": Angle(0.0, u.deg)}
-
-
-def _sun_pos(date):
-    """
-    Calculate solar ephemeris parameters.  Allows for planetary and lunar
-    perturbations in the calculation of solar longitude at date and various
-    other solar positional parameters. This routine is a truncated version of
-    Newcomb's Sun and is designed to give apparent angular coordinates (T.E.D)
-    to a precision of one second of time.  This function replicates the SSW/
-    IDL function "sun_pos.pro".  This function is assigned to be
-    internal at the moment as it should really be replaced by accurate
-    ephemeris calculations in the part of SunPy that handles ephemeris.
-
-    Parameters
-    -----------
-    date : `sunpy.time.time`
-        Time at which the solar ephemeris parameters are calculated.  The
-        input time can be in any acceptable time format.
-
-    Returns
-    -------
-    A dictionary with the following keys with the following meanings:
-
-    longitude  -  Longitude of sun for mean equinox of date (degs)
-    ra         -  Apparent RA for true equinox of date (degs)
-    dec        -  Apparent declination for true equinox of date (degs)
-    app_long   -  Apparent longitude (degs)
-    obliq      -  True obliquity (degs)
+    # Compute Stonyhurst Heliographic co-ordinates - returns (longitude,
+    # latitude). Points off the limb are returned as nan.
+    heliographic_coordinate = coordinate.transform_to('heliographic_stonyhurst')
 
-    Notes
-    -----
-    SSWIDL code equivalent:
-        http://hesperia.gsfc.nasa.gov/ssw/gen/idl/solar/sun_pos.pro
+    # Compute the differential rotation
+    drot = diff_rot(interval, heliographic_coordinate.lat.to(u.degree), **diff_rot_kwargs)
+
+    # Set the observer
+    if new_observer_location is None:
+        observer = get_earth(time=new_observer_time)
+    elif isinstance(new_observer_location, six.string_types):
+        if new_observer_location.lower() == 'earth':
+            observer = get_earth(time=new_observer_time)
+        else:
+            observer = get_body_heliographic_stonyhurst(new_observer_location, time=new_observer_time)
+    elif isinstance(new_observer_location, BaseCoordinateFrame):
+        observer = new_observer_location
+    elif isinstance(new_observer_location, SkyCoord):
```

Comment:
you don't need to do this, it is ok to pass a `SkyCoord` as the observer attribute to a frame (it will be converted to a frame internally).